plugin-gradle/src/main/java/com/diffplug/gradle/spotless/JavaExtension.java

```diff
@@ -83,16 +83,23 @@ public GoogleJavaFormatConfig googleJavaFormat() {
 	 * for an workaround for using snapshot versions.
 	 */
 	public GoogleJavaFormatConfig googleJavaFormat(String version) {
+		return googleJavaFormat(GoogleJavaFormatStep.defaultArtifact(), version);
+	}
+
+	public GoogleJavaFormatConfig googleJavaFormat(String artifact, String version) {
```

Comment:
This method should not exist

Reply:
The version might be not consistent with the official GJF. Since the artifact is home-brewed, the version should be specified together.

Reply:
Valid point. Technically, a [maven coordinate](https://maven.apache.org/pom.html#Maven_Coordinates) is `groupId:artifactId:version`. How about `googleJavaFormat().mavenCoordinate('com.acme:gfj-hack:3.0')`? The maven DSL is not as flexible, but you could assert that if `mavenCoordinate` is non-null then `version` must be the default, and make a nice error message about "If you set mavenCoordinate then you must not set the version."

Reply:
Probably `customMavenCoordinate`, this is a rare usecase...

Reply:
I revisit all the logic related to the `version`. There are some bugfixes bound to it. I would not change those in this PR. So I'll go for your suggestions.